src/c/c_client.cpp

```diff
@@ -30,517 +30,654 @@
-  return (void*)dataset;
+  // Sanity check params
+  if (NULL == c_client || NULL == name)
+    return NULL; // Nothing we can do, so just bail. (error reporting to come later)
+
+  Client *s = reinterpret_cast<Client *>(c_client);
+  std::string dataset_name(name, name_length);
+  DataSet *dataset = NULL;
+
+  try {
+    dataset = new DataSet(s->get_dataset(dataset_name));
+  } catch (...) {
+    dataset = NULL;
+  }
+  return reinterpret_cast<void *>(dataset);
 }
 
+// Rename a dataset in the database.
 extern "C"
-void rename_dataset(void* c_client, const char* name,
-                    const size_t name_length, const char* new_name,
+void rename_dataset(void *c_client, const char *name,
+                    const size_t name_length, const char *new_name,
                     const size_t new_name_length)
 {
-  /* Rename a dataset in the database.
-  */
-  Client* s = (Client *)c_client;
-  std::string name_str = std::string(name, name_length);
-  std::string new_name_str = std::string(new_name, new_name_length);
+  // Sanity check params
+  if (NULL == c_client || NULL == name || NULL == new_name)
+    return; // Nothing we can do, so just bail. (error reporting to come later)
+
+  Client *s = reinterpret_cast<Client *>(c_client);
+  std::string name_str(name, name_length);
+  std::string new_name_str(new_name, new_name_length);
+
   s->rename_dataset(name_str, new_name_str);
-  return;
 }
 
+// Copy a dataset from the src_name to the dest_name
 extern "C"
-void copy_dataset(void* c_client, const char* src_name,
-                  const size_t src_name_length, const char* dest_name,
-                  const size_t dest_name_length
-                  )
+void copy_dataset(void *c_client, const char *src_name,
+                  const size_t src_name_length, const char *dest_name,
+                  const size_t dest_name_length)
 {
-  /* Copy a dataset from teh src_name to the dest_name
-  */
-  Client* s = (Client *)c_client;
-  std::string src_name_str = std::string(src_name, src_name_length);
-  std::string dest_name_str = std::string(dest_name, dest_name_length);
+  // Sanity check params
+  if (NULL == c_client || NULL == src_name || NULL == dest_name)
+    return; // Nothing we can do, so just bail. (error reporting to come later)
+
+  Client *s = reinterpret_cast<Client *>(c_client);
+  std::string src_name_str(src_name, src_name_length);
+  std::string dest_name_str(dest_name, dest_name_length);
+
   s->copy_dataset(src_name_str, dest_name_str);
-  return;
 }
 
+// Delete a dataset (all metadata and tensors) from the database.
 extern "C"
-void delete_dataset(void* c_client, const char* name,
-                    const size_t name_length)
+void delete_dataset(void *c_client, const char *name, const size_t name_length)
 {
-  /* Delete a dataset (all metadata and tensors) from the
-  database.
-  */
-  Client* s = (Client *)c_client;
-  std::string dataset_name = std::string(name, name_length);
+  // Sanity check params
```

Comment:
I wonder if this ``// Sanity check params`` comment is needed throughout the code or we just remove.  It seems intuitive what is being checked.

Reply:
Extra comments don't hurt. I'd rather see more comments in the code rather than less.